### PR TITLE
update for ipywidgets 8.0

### PR DESCRIPTION
**Pull request by akaszynski**
This PR temporarly limits the version if ipywidgets to `<8.0` until https://github.com/vidartf/ipydatawidgets/pull/47 is merged and a new release of `ipydatawidgets` is released.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -84,10 +84,18 @@
...
+            'cmocean',
+            'meshio>=5.2',
+            'ipywidgets<8.0',
+            'ipyvtklink',
+            'pythreejs',
+        ],
         'colormaps': ['matplotlib', 'colorcet', 'cmocean'],
         'io': ['meshio>=5.2'],
-        'jupyter': ['ipyvtklink', 'pythreejs'],
+        'jupyter': ['ipyvtklink', 'ipywidgets<8.0', 'panel' 'pythreejs'],
```
Comment (akaszynski) [inline setup.py:98]:
```suggestion
        'jupyter': ['ipyvtklink', 'ipywidgets<8.0', 'panel', 'pythreejs'],
```

File: pyvista/jupyter/pv_ipygany.py [hunk 2]
```diff
@@ -194,7 +194,8 @@ def show_ipygany(plotter, return_viewer, height=None, width=None):
 
     if cbar is not None:
         # Colormap choice widget
-        colormap_dd = Dropdown(options=colormaps, description='Colormap:')
+        options = [(key, value) for key, value in colormaps.items()]
+        colormap_dd = Dropdown(options=options, description='Colormap:')
```
Comment (akaszynski) [inline pyvista/jupyter/pv_ipygany.py:198]: This is needed for `ipywidgets==8.0` when we do eventually support it.

File: pyvista/jupyter/pv_ipygany.py [hunk 3]
```diff
@@ -194,7 +194,8 @@ def show_ipygany(plotter, return_viewer, height=None, width=None):
 
     if cbar is not None:
         # Colormap choice widget
-        colormap_dd = Dropdown(options=colormaps, description='Colormap:')
+        options = [(key, value) for key, value in colormaps.items()]
```
Comment (MatthewFlamm) [inline pyvista/jupyter/pv_ipygany.py:197]: This is the same as `options = list(colormaps.items())`?

File: setup.py [hunk 4]
```diff
@@ -84,10 +84,18 @@
...
+            'cmocean',
+            'meshio>=5.2',
+            'ipywidgets<8.0',
+            'ipyvtklink',
+            'pythreejs',
+        ],
         'colormaps': ['matplotlib', 'colorcet', 'cmocean'],
         'io': ['meshio>=5.2'],
-        'jupyter': ['ipyvtklink', 'pythreejs'],
+        'jupyter': ['ipyvtklink', 'ipywidgets<8.0', 'panel', 'pythreejs'],
```
Comment (MatthewFlamm) [inline setup.py:98]: Shouldn't `'all'` now also include `'panel'`?

**Review by MatthewFlamm — COMMENTED** [review]
This should be a priority as we wait on changes upstream.

File: pyvista/jupyter/pv_ipygany.py (hunk 3 above)
Comment (akaszynski) [reply pyvista/jupyter/pv_ipygany.py:197]:
```suggestion
        options = list(colormaps.items())
```

File: setup.py [hunk 5]
```diff
@@ -84,10 +84,18 @@
...
     extras_require={
-        'all': ['matplotlib', 'colorcet', 'cmocean', 'meshio>=5.2', 'ipyvtklink', 'pythreejs'],
+        'all': [
+            'matplotlib',
+            'colorcet',
+            'cmocean',
+            'meshio>=5.2',
+            'ipywidgets<8.0',
+            'ipyvtklink',
+            'pythreejs',
```
Comment (akaszynski) [inline setup.py:94]:
```suggestion
            'panel',
            'pythreejs',
```

File: setup.py (hunk 4 above)
Comment (akaszynski) [reply setup.py:98]: Yep. Added.

**Comment by akaszynski** [comment]
> Build Documentation / Build Documentation (pull_request) Successful in 15m

This makes me so happy.

**Comment by akaszynski** [comment]
Merging this as it's a blocker and people have commented. Thanks for the review @MatthewFlamm.

**Pull request merged**